pkg/storage/database/models/models.go

```diff
@@ -62,9 +64,14 @@ type ConnectionRequest struct {
 
 type APIKey struct {
 	gorm.Model
+	Name          string
 	DestinationID uint
 	Destination   Destination `gorm:"constraint:OnDelete:CASCADE"`
 	HashedAPIKey  string      `gorm:"index"`
+	// If the API key is specific to a saved query
+	SavedQueryID uint
```

Comment:
One comment here (I realize this is copied from my existing branch.) I'd started to refactor this so that the APIKey table does not have a SavedQuery, and instead use a Join table, ie, `apikey_savedquery`. Then I just added an APIKey "type" (`QUERY`, `GENERAL` - or something more descriptive) that tells me about who the key belongs to.

This also then lets me create a single API key and give it access to multiple endpoints. And it didn't make sense the `QueryParams` be in this table either since it's such a specific use case.